### PR TITLE
Grpc client streaming

### LINKED ISSUE #1: Support for MethodType.CLIENT_STREAMING
Any plans for supporting `io.grpc.MethodDescriptor.MethodType.CLIENT_STREAMING` rpc calls?

### PR DESCRIPTION
**Pull request by queimadus**
Adds support for `MethodType.CLIENT_STREAMING`.

Fixes #1.

### CODE REVIEW

**Comment by queimadus** [comment]
@eiennohito, PTAL and leave feedback when you have time :smile:

File: src/main/scala/org/eiennohito/grpc/stream/impl/client/GrpcClientHandler.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ class GrpcClientHandler[Req, Resp](
       override def onPush(): Unit = {
         val msg = grab(in)
         call.sendMessage(msg)
-        if (isAvailable(in)) {
+        if (!hasBeenPulled(in) && call.isReady) {
```
Comment (eiennohito) [inline src/main/scala/org/eiennohito/grpc/stream/impl/client/GrpcClientHandler.scala:54]: nice find! This was a bug.

**Comment by eiennohito** [comment]
Could you please add tests for interop of JavaApi <-> AkkaStreamApi

**Comment by queimadus** [comment]
Interoperability test added in 1654224.

**Pull request merged**